osdk/src/commands/test.rs

```diff
@@ -6,10 +6,18 @@ use super::{build::do_build, util::DEFAULT_TARGET_RELPATH};
 use crate::{
     base_crate::new_base_crate,
     config_manager::{BuildConfig, RunConfig, TestConfig},
-    util::{get_current_crate_info, get_target_directory},
+    util::{get_cargo_metadata, get_current_crate_info, get_target_directory},
 };
 
 pub fn execute_test_command(config: &TestConfig) {
+    let crates = get_workspace_default_members();
+    for crate_path in crates {
```

Comment:
I didn't get it clear. 

From https://doc.rust-lang.org/cargo/commands/cargo-test.html#package-selection,
> If the manifest is the root of a workspace then the workspaces default members are selected, otherwise only the package defined by the manifest will be selected.

So if running `cargo osdk test` in workspace root, we should go to all default members and run test, right?

Reply:
Ah, what I see is that the Makefile is still controlling what crates to test using what tools. the --workspace --exclude option can be used to rule out the bzimage-setup crate, which can't be tested right now.

Reply:
Yes, the test interface can be refined once we have per-crate's manifest.